mantle/cmd/kola/testiso.go

```diff
@@ -233,22 +271,62 @@ RequiredBy=coreos-installer.target
 RequiredBy=multi-user.target`, nmConnectionId, nmConnectionFile)
 
 func init() {
+	archPatterns = getPatterns()
 	cmdTestIso.Flags().BoolVarP(&instInsecure, "inst-insecure", "S", false, "Do not verify signature on metal image")
-	cmdTestIso.Flags().BoolVarP(&nopxe, "no-pxe", "P", false, "Skip testing live installer PXE")
-	cmdTestIso.Flags().BoolVarP(&noiso, "no-iso", "", false, "Skip testing live installer ISO")
 	cmdTestIso.Flags().BoolVar(&console, "console", false, "Connect qemu console to terminal, turn off automatic initramfs failure checking")
 	cmdTestIso.Flags().BoolVar(&pxeAppendRootfs, "pxe-append-rootfs", false, "Append rootfs to PXE initrd instead of fetching at runtime")
 	cmdTestIso.Flags().StringSliceVar(&pxeKernelArgs, "pxe-kargs", nil, "Additional kernel arguments for PXE")
-	cmdTestIso.Flags().BoolVar(&addNmKeyfile, "add-nm-keyfile", false, "Add NetworkManager connection keyfile")
-	cmdTestIso.Flags().StringSliceVar(&scenarios, "scenarios", []string{scenarioPXEInstall, scenarioISOOfflineInstall, scenarioPXEOfflineInstall, scenarioISOLiveLogin, scenarioISOAsDisk, scenarioMinISOInstall, scenarioMinISOInstallNm}, fmt.Sprintf("Test scenarios (also available: %v)", []string{scenarioISOInstall}))
+	cmdTestIso.Flags().StringSliceVar(&patterns, "patterns", archPatterns, "Test patterns available")
```

Comment:
In the way it is, it will already run all tests by default it we don't pass any test.

Reply:
> In the way it is, it will already run all tests by default it we don't pass any test.

Just following up on this; to expand, I think we do want the default behaviour you describe. But when we do want to run one or more specific test(s), I think we should match the `kola run` semantics. E.g. we don't type `kola run --tests basic`, we just run `kola run basic`. But cool to keep this as a follow-up!